build failure: mdk4

### Nixpkgs version

- Unstable (25.05)

### Describe the bug

mdk4 doesn't compile

### Steps to reproduce

use mdk4 in latest nixos

### Expected behaviour

for it to compile

### Screenshots

_No response_

### Relevant log output

```console
error: builder for '/nix/store/zcdgkqrbdq0s4iqh2a33j6g79p8xaw7n-mdk4-unstable-2021-04-27.drv' failed with exit code 2;
       last 25 log lines:
       >   155 | static int error_handler(struct sockaddr_nl *nla, struct nlmsgerr *err,
       >       |                          ~~~~~~~~~~~~~~~~~~~~^~~
       > channelhopper.c: In function 'finish_handler':
       > channelhopper.c:163:42: warning: unused parameter 'msg' []
       >   163 | static int finish_handler(struct nl_msg *msg, void *arg)
       >       |                           ~~~~~~~~~~~~~~~^~~
       > channelhopper.c: In function 'ack_handler':
       > channelhopper.c:170:39: warning: unused parameter 'msg' []
       >   170 | static int ack_handler(struct nl_msg *msg, void *arg)
       >       |                        ~~~~~~~~~~~~~~~^~~
       > channelhopper.c: In function 'valid_handler':
       > channelhopper.c:186:45: warning: unused parameter 'arg' []
       >   186 | int valid_handler(struct nl_msg *msg, void *arg)
       >       |                                       ~~~~~~^~~
       > channelhopper.c: In function 'handle_channels':
       > channelhopper.c:328:50: warning: unused parameter 'state' []
       >   328 | static int handle_channels(struct nl80211_state *state, struct nl_msg *msg)
       >       |                            ~~~~~~~~~~~~~~~~~~~~~~^~~~~
       > channelhopper.c: At top level:
       > channelhopper.c:240:14: warning: 'dfs_state_name' defined but not used []
       >   240 | static char *dfs_state_name(enum nl80211_dfs_state state)
       >       |              ^~~~~~~~~~~~~~
       > make[1]: *** [<builtin>: channelhopper.o] Error 1
       > make[1]: Leaving directory '/build/source/src'
       > make: *** [Makefile:10: all] Error 2
       For full logs, run 'nix log /nix/store/zcdgkqrbdq0s4iqh2a33j6g79p8xaw7n-mdk4-unstable-2021-04-27.drv'.
```

### Additional context

_No response_

### System metadata

nako@nixos /e/nixos [1]> nix-shell -p nix-info --run "nix-info -m"                                                                            (base) 
 - system: `"x86_64-linux"`
 - host os: `Linux 6.13.3, NixOS, 25.05 (Warbler), 25.05.20250218.73cf49b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.12`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/kdynjy1mbgkdg4p196v9gx6ljpf7q4nk-source`

### Notify maintainers


---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/md/mdk4/package.nix
+++ b/pkgs/by-name/md/mdk4/package.nix
@@ -9,13 +9,13 @@
 
 stdenv.mkDerivation {
   pname = "mdk4";
-  version = "unstable-2021-04-27";
+  version = "4.2-unstable-2024-08-16";
 
   src = fetchFromGitHub {
     owner = "aircrack-ng";
     repo = "mdk4";
-    rev = "e94422ce8e4b8dcd132d658345814df7e63bfa41";
-    sha256 = "sha256-pZS7HQBKlSZJGqoZlSyBUzXC3osswcB56cBzgm+Sbwg=";
+    rev = "36ca143a2e6c0b75b5ec60143b0c5eddd3d2970c";
+    hash = "sha256-iwESQgvt9gLQeDKVkf9KcztQmjdCLOE0+Q0FlfbbjEU=";
   };
 
   preBuild = ''
@@ -37,11 +37,13 @@ stdenv.mkDerivation {
     "SBINDIR=$(PREFIX)/bin"
   ];
 
-  meta = with lib; {
+  enableParallelBuilding = true;
+
+  meta = {
     description = "Tool that injects data into wireless networks";
     homepage = "https://github.com/aircrack-ng/mdk4";
-    maintainers = with maintainers; [ moni ];
-    license = licenses.gpl2Plus;
+    maintainers = with lib.maintainers; [ moni ];
+    license = lib.licenses.gpl3Plus;
     mainProgram = "mdk4";
   };
 }

